src/System.Console/src/System/ConsolePal.WinRT.cs

```diff
@@ -0,0 +1,256 @@
+
+        internal sealed class ControlCHandlerRegistrar
+        {
+            internal void Register() { throw new PlatformNotSupportedException(); }
+
+            internal void Unregister() { throw new PlatformNotSupportedException(); }
+        }
+
+        public static Stream OpenStandardInput() { throw new PlatformNotSupportedException(); }
+
+
+        public static Stream OpenStandardOutput() { return new NoOpStream(FileAccess.Write); }
+
+        public static Stream OpenStandardError() { return new NoOpStream(FileAccess.Write); }
+
+        public static Encoding InputEncoding
+        {
+            get { return new UTF8Encoding(encoderShouldEmitUTF8Identifier: false); }
+        }
+
+        public static Encoding OutputEncoding
+        {
+            get { return new UTF8Encoding(encoderShouldEmitUTF8Identifier: false); }
+        }
+
+        public static bool KeyAvailable { get { throw new PlatformNotSupportedException(); } }
+
+        public static ConsoleKeyInfo ReadKey(bool intercept)
+        {
+            if (Console.IsInputRedirected)
+            {
+                // We could leverage Console.Read() here however
+                // windows fails when stdin is redirected.
+                throw new InvalidOperationException(SR.InvalidOperation_ConsoleReadKeyOnFile);
+            }
+            throw new PlatformNotSupportedException();
+        }
+
+        public static bool TreatControlCAsInput
+        {
+            get
+            {
+                if (Console.IsInputRedirected)
+                    return false;
+                throw new PlatformNotSupportedException();
+            }
+            set
+            {
+                if (!Console.IsInputRedirected)
+                    throw new InvalidOperationException(SR.InvalidOperation_ConsoleReadKeyOnFile);
+            }
+        }
+
+        public static ConsoleColor ForegroundColor
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        public static ConsoleColor BackgroundColor
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        public static void ResetColor() { throw new PlatformNotSupportedException(); }
+
+        public static bool NumberLock { get { throw new PlatformNotSupportedException(); } }
+
+        public static bool CapsLock { get { throw new PlatformNotSupportedException(); } }
+
+        public static int CursorSize
+        {
+            get { return 100; }
+            set { throw new PlatformNotSupportedException(); }
+        }
+
+        public static string Title
+        {
+            get { throw new PlatformNotSupportedException(); }
+            set
+            {
+                if (!Console.IsOutputRedirected)
```

Comment:
>Why are we special-casing redirection like this?

Unix ignores a bunch of these operations when output/input are redirected so it seemed logical that WinRT should function similarly. Since there was precedence for making the operations no-op, I followed it.

>Especially since it looks like we throw a PlatformNotSupportedException from IsOutput/InputRedirected....?

That's an oversight on my part. I didn't notice there was a `_isStdOutRedirected` **and** a `s_isOutTextWriterRedirected` that were used in different places. The idea was that most of those operations would no-op when someone called SetOut before calling them, but calling SetOut only modifies `s_isOutTextWriterRedirected`. Since we have no notion of console handle redirection in UWP (since there isn't a console handle), I was hoping to approximate that behavior with the TextWriter redirection. Thinking more about it, that's too big of a stretch and diverges from our existing implementations more than I'd like, so I'm going to remove these Redirected checks.